### ISSUE TITLE
DNS check on install fails on minikube

### ISSUE THREAD

**Issue by foot** [weaveworks/launcher#285]
```
curl -Ls https://get.weave.works |
>   sh -s -- --token=XXXXX
Downloading the Weave Cloud installer from https://get.weave.works/bootstrap?dist=darwin
Preparing for Weave Cloud setup
Checking kubectl & kubernetes versions
Connecting cluster to "Simon minikube test June" (id: still-water-52) on Weave Cloud
Installing Weave Cloud agents on launcher-tests at https://192.168.99.100:8443
Performing a check of the Kubernetes installation setup.
There was an error while performing a DNS check: checking DNS failed, the DNS \
in the Kubernetes cluster is not working correctly. \
Please check that your cluster can download images and run pods.
```

The command it is running is to test DNS: https://github.com/weaveworks/launcher/blob/master/pkg/kubectl/kubectl.go#L265 , e.g.:

```bash
$ kubectl run --image docker.io/busybox --labels=launcher=dns \
    --restart=Never --command "test-dns-pod" nslookup cloud.weave.works`

$ kubectl get pod test-dns-pod
NAME           READY   STATUS   RESTARTS   AGE
test-dns-pod   0/1     Error    0          10s

$ kubectl logs test-dns-pod
Server:		10.96.0.10
Address:	10.96.0.10:53

** server can't find cloud.weave.works: NOTIMP

*** Can't find cloud.weave.works: No answer
```

busybox 1.30 actually also fails but doesn't return a bad exit code /o\... (_note STATUS: Completed_)

```bash
$ kubectl run --image docker.io/busybox:1.30 --labels=launcher=dns \
    --restart=Never --command "test-dns-pod" nslookup cloud.weave.works

$ kubectl get pod test-dns-pod
NAME           READY   STATUS      RESTARTS   AGE
test-dns-pod   0/1     Completed   0          2m

$ kubectl logs test-dns-pod
Server:		10.96.0.10
Address:	10.96.0.10:53

** server can't find cloud.weave.works: NOTIMP

*** Can't find cloud.weave.works: No answer
```

Alpine actually works:
```bash
$ kubectl run --image alpine --labels=launcher=dns \
    --restart=Never --command "test-dns-pod" nslookup cloud.weave.works

$ kubectl get pod test-dns-pod
NAME           READY   STATUS      RESTARTS   AGE
test-dns-pod   0/1     Completed   0          13s

$ kubectl logs test-dns-pod
nslookup: can't resolve '(null)': Name does not resolve

Name:      cloud.weave.works
Address 1: 34.202.121.123 ec2-34-202-121-123.compute-1.amazonaws.com
Address 2: 34.197.195.29 ec2-34-197-195-29.compute-1.amazonaws.com
Address 3: 54.85.235.49 ec2-54-85-235-49.compute-1.amazonaws.com
```

**Comment by foot** [comment]
- busybox:1.28 works, 1.29 not.

**Comment by foot** [comment]
If I give it the IP of the coredns pod exlicitly it works:
```bash
$ kubectl get pods -owide -A
NAMESPACE     NAME                               READY   STATUS      RESTARTS   AGE   IP           NODE       NOMINATED NODE   READINESS GATES
default       test-dns-pod                       0/1     Completed   0          28s   172.17.0.4   minikube   <none>           <none>
kube-system   coredns-fb8b8dccf-bcxll            1/1     Running     0          50m   172.17.0.2   minikube   <none>           <none>
kube-system   coredns-fb8b8dccf-fwxrr            1/1     Running     0          50m   172.17.0.3   minikube   <none>           <none>
kube-system   etcd-minikube                      1/1     Running     0          49m   10.0.2.15    minikube   <none>           <none>
kube-system   kube-addon-manager-minikube        1/1     Running     0          49m   10.0.2.15    minikube   <none>           <none>
kube-system   kube-apiserver-minikube            1/1     Running     0          49m   10.0.2.15    minikube   <none>           <none>
kube-system   kube-controller-manager-minikube   1/1     Running     0          49m   10.0.2.15    minikube   <none>           <none>
kube-system   kube-proxy-6xlt9                   1/1     Running     0          50m   10.0.2.15    minikube   <none>           <none>
kube-system   kube-scheduler-minikube            1/1     Running     0          49m   10.0.2.15    minikube   <none>           <none>
kube-system   storage-provisioner                1/1     Running     0          50m   10.0.2.15    minikube   <none>           <none>

$ kubectl run --image docker.io/busybox --labels=launcher=dns \
    --restart=Never --command "test-dns-pod" nslookup cloud.weave.works 172.17.0.2

$ kubectl logs test-dns-pod
Server:		172.17.0.2
Address:	172.17.0.2:53

Non-authoritative answer:
cloud.weave.works	canonical name = frontend.prod.weave.works
Name:	frontend.prod.weave.works
Address: 54.85.235.49
Name:	frontend.prod.weave.works
Address: 34.197.195.29
Name:	frontend.prod.weave.works
Address: 34.202.121.123

*** Can't find cloud.weave.works: No answer
```

**Comment by foot** [comment]
@dlespiau do you wanna do any further poking? A quick fix is a downgrade to busybox 1.28 which is the version that pops up in the k8s docs: https://kubernetes.io/docs/tasks/administer-cluster/dns-debugging-resolution/
